src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -218,6 +291,67 @@ private static unsafe bool TryEncodeControlChar(SymbolTable symbolTable, byte va
             return symbolTable.TryEncode(value, buffer, out written);
         }
 
+        private static int GetHashCode(ReadOnlySpan<byte> span)
+        {
+            int hash = 17;
+            foreach (byte element in span)
+            {
+                hash = hash * 31 + element;
+            }
+            return hash;
+        }
+
+        private static Dictionary<int, PropertyInfoLinkedList> GetTypeMap(Type type)
+        {
+            var dictionary = new Dictionary<int, PropertyInfoLinkedList>();
+            IEnumerable<PropertyInfo> properties = type.GetRuntimeProperties();
+            foreach (PropertyInfo pi in properties)
+            {
+                byte[] encoded = Encoding.UTF8.GetBytes(pi.Name);
+                int hashcode = GetHashCode(encoded);
+
+                if (dictionary.ContainsKey(hashcode))
+                {
+                    dictionary.TryGetValue(hashcode, out PropertyInfoLinkedList list);
```

Comment:
why do you map from hashcodes? and then store lists? This is what dictionary is doing again (in it's implementation).

Reply:
This (hashcode, PropertyInfoList) mapping is needed since our json reader returns a `ReadOnlySpan<byte>` as the value which cannot be stored in a collection without allocating. 

So, from the input type T, we have a list of property names on the type (which are string). For fast access of the property info that matches the property name, I calculate the hashcode of the `ReadOnlySpan<byte>` returned by the reader and use that for the look up of the property info. Since it is possible for their to be a hash conflict (where two properties end up with the same hashcode), our dictionary needs to be (int, PropertyInfoList) rather than (int, PropertyInfo). In most cases, the list would only contain a single PropertyInfo anyway.